docs_src/design/ucr/Microservice-Authentication.md

```diff
@@ -0,0 +1,145 @@
+  with applicable security policies and procedures._
+
+- [Critical Manufacturing Sector Cybersecurity Framework Implementation Guidance](https://www.cisa.gov/sites/default/files/publications/Critical_Manufacturing_Sector_Cybersecurity_Framework_Implementation_Guidance_FINAL_508.pdf)
+
+  _PR.AC-1: Identities and credentials are issued, managed, verified,
+  revoked, and audited for authorized devices, users, and processes._
+
+
+### Target Users
+- Device Owner
+- Device User
+- Device Maintainer
+- Service Provider
+
+### Description
+
+Microservice authentication provides the following benefits,
+which are potentially valuable to all of the listed target users:
+
+- Provides a defense against malware running on the device,
+  as currently there is no mechanism to ensure that only
+  authorized users or processes are allowed to invoke EdgeX services.
+
+- Provides greater auditability as to who initiated a particular
+  action on the device.
+
+- Depending on implementation, may provide a way to revoke
+  access that was previously granted,
+  or allow customers to tie in to enterprise
+  identity management systems.
+
+For purposes of this UCR, microservice authentication implies
+that the receiving microservice has access to the identity
+of the caller and can write program logic based on that identity.
+
+
+### Existing solutions
+
+Microservice authentication is currently implemented around two primary vectors:
+
+- Token-based authentication schemes.
+  Initiator sends an identifier along with a request to the receiver.
+  The identifier is cryptographically validated using a key trusted by
+  the receiver, or the receiver asks a trusted third party to verify the identifier.
+
+  A benefit of token-based authentication schemes is identity delegation,
+  whereby the identifier can be passed through a chain of calls to
+  preserve the identity of the original initiator.
+  The identifier can often be tunneled through other protocols.
+  Another benefit of token-based authentication is that
+  it flows easily through a web application firewall.
+
+  A drawback of token-based authentication is that due to MITM threats,
+  token-based authentication over an unencrypted network is insecure.
+  Another drawback of token-based authentication is that it is unidirectional:
+  the receiver can authenticate the initiator, but not vice-versa.
+
+- End-to-end encryption schemes.
+  Both the initiator and the receiver participate in a session-oriented message exchange over an encrypted transport,
+  where both parties cryptographically validate each other's identity.
+  
+  A benefit of end-to-end encryption schemes is that it enforces proof-of-possession
+  (of a cryptographic key), thus preventing token-stealing attacks.
+
+  End-to-end encryption schemes are strongest when implemented at the application level,
+  as it does not require trust in the underlying network,
+  with the potential drawback that if network-level encryption is used,
```

Comment:
Without trying to split hairs here, I'd like to just provide some context from my perspective...

There are a few different layers of encryption depending on implementation. OpenZiti for example uses mTLS between overlay network nodes, uses e2ee for the initiator/terminator of overlay traffic, and then if your protocol itself is encrypted (https/scp/etc) you'll end up with yet another tier of encryption. In that situation the only 'redundant' encryption would be the e2ee but in my opinion, all encryption is redundant, until it isn't, right? :)

So in that situation network level, or "link" level encryption is definitely used which prevents a bad actor from MITM on a router. e2ee is used to prevent a bad network operator from using encryption keys for the mTLS, and application encryption would be used to ensure the e2ee encryption is not needed to be trusted.

Reply:
See next update.